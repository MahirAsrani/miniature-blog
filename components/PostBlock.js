/* eslint-disable @next/next/no-img-element */
import Link from 'next/link';
import styles from '../styles/PostBlock.module.css';

const PostBlock = ({
  id,
  title,
  slug,
  featureImg,
  content,
  author,
  postedDate,
  category,
  categorySlug,
}) => {
  const date = new Date(postedDate).toDateString();

  return (
    <div className="col-md-6 col-lg-4">
      <div className="position-relative">
        <div
          className={styles.card_img_div}
          style={{
            backgroundImage: `url(${featureImg || './uploads/No_Image.png'})`,
          }}
        ></div>
        <div
          className={`card-img-overlay d-flex align-items-start flex-column p-3 ${styles.graident}`}
        >
          <div className={`w-100 mt-auto ${styles.z5}`}>
            <a href={`/${categorySlug}`} className="badge bg-warning mb-2">
              <i className="fas fa-circle me-2 small fw-bold"></i>
              {category}
            </a>
          </div>
        </div>
      </div>
      <div className={`${styles.card_body} px-0 py-3`}>
        <h4 className="card-title">
          <Link href={`/${categorySlug}/${slug}`}>
            <a data-testid="title" className="btn-link text-reset fw-bold">
              {title}
            </a>
          </Link>
        </h4>
        <p className="card-text" data-testid="content">
          {content.length > 30 ? content.substring(0, 100) + '...' : content}
        </p>

        <ul className="nav align-items-center d-flex">
          <li className="nav-item">
            <div className="nav-link ps-0">
              <div className="d-flex align-items-center position-relative">
                <div className={styles.avatar}>
                  <img
                    className="avatar-img rounded-circle"
                    src={author.image || '/uploads/profile/no_image.jpg'}
                    alt="avatar"
                    data-testid="author-img"
                  />
                </div>
                <span className="ms-2">
                  <Link href="#">
                    <a
                      className="stretched-link text-reset btn-link"
                      data-testid="author-name"
                    >
                      by {author.name}
                    </a>
                  </Link>
                </span>
              </div>
            </div>
          </li>
          <li className="nav-item" data-testid="date">
            {date}
          </li>
        </ul>
      </div>
    </div>
  );
};

export default PostBlock;
