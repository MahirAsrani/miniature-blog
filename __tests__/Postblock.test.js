import '@testing-library/jest-dom/extend-expect';
import { render, screen } from '@testing-library/react';
import PostBlock from '../components/PostBlock';

describe('PostBlock Component', () => {
  it('renders homepage unchanged', () => {
    const date = new Date();
    const postedDate = new Date(date).toDateString();

    render(
      <PostBlock
        id="1"
        title="Title"
        slug="title"
        featureImg={null}
        content="content"
        author={{
          image: null,
          name: 'Person',
        }}
        postedDate={date}
        category="Tech"
        categorySlug="tech"
      />
    );

    expect(screen.getByTestId('title')).toHaveTextContent('Title');
    expect(screen.getByTestId('content')).toHaveTextContent('content');
    expect(screen.getByTestId('date')).toHaveTextContent(postedDate);
    expect(screen.getByTestId('author-name')).toHaveTextContent('Person');
    expect(screen.getByTestId('author-img')).toBeInTheDocument();
  });

  it('renders unchanged', () => {
    const date = new Date();

    const { container } = render(
      <PostBlock
        id="1"
        title="Title"
        slug="title"
        featureImg={null}
        content="content"
        author={{
          image: null,
          name: 'Person',
        }}
        postedDate={date}
        category="Tech"
        categorySlug="tech"
      />
    );
    expect(container).toMatchSnapshot();
  });
});
