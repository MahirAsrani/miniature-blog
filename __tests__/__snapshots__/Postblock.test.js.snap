// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostBlock Component renders unchanged 1`] = `
<div>
  <div
    class="col-md-6 col-lg-4"
  >
    <div
      class="position-relative"
    >
      <div
        class="card_img_div"
        style="background-image: url(./uploads/No_Image.png);"
      />
      <div
        class="card-img-overlay d-flex align-items-start flex-column p-3 graident"
      >
        <div
          class="w-100 mt-auto z5"
        >
          <a
            class="badge bg-warning mb-2"
            href="/tech"
          >
            <i
              class="fas fa-circle me-2 small fw-bold"
            />
            Tech
          </a>
        </div>
      </div>
    </div>
    <div
      class="card_body px-0 py-3"
    >
      <h4
        class="card-title"
      >
        <a
          class="btn-link text-reset fw-bold"
          data-testid="title"
          href="/tech/title"
        >
          Title
        </a>
      </h4>
      <p
        class="card-text"
        data-testid="content"
      >
        content
      </p>
      <ul
        class="nav align-items-center d-flex"
      >
        <li
          class="nav-item"
        >
          <div
            class="nav-link ps-0"
          >
            <div
              class="d-flex align-items-center position-relative"
            >
              <div
                class="avatar"
              >
                <img
                  alt="avatar"
                  class="avatar-img rounded-circle"
                  data-testid="author-img"
                  src="/uploads/profile/no_image.jpg"
                />
              </div>
              <span
                class="ms-2"
              >
                <a
                  class="stretched-link text-reset btn-link"
                  data-testid="author-name"
                  href="/#"
                >
                  by 
                  Person
                </a>
              </span>
            </div>
          </div>
        </li>
        <li
          class="nav-item"
          data-testid="date"
        >
          Mon Jun 27 2022
        </li>
      </ul>
    </div>
  </div>
</div>
`;
